autometacal/python/metacal.py

```diff
@@ -83,3 +83,58 @@ def get_metacal_response(gal_image,
   R = tape.batch_jacobian(e, g)
   return e, R
 
+def get_metacal_response_finitediff(gal_image,psf_image,reconv_psf_image,step,method):
+  
+  img0s = autometacal.generate_mcal_image(gal_image,psf_image,reconv_psf_image,[[0,0]]) 
+  img1p = autometacal.generate_mcal_image(gal_image,psf_image,reconv_psf_image,[[step,0]]) 
+  img1m = autometacal.generate_mcal_image(gal_image,psf_image,reconv_psf_image,[[-step,0]]) 
+  img2p = autometacal.generate_mcal_image(gal_image,psf_image,reconv_psf_image,[[0,step]]) 
+  img2m = autometacal.generate_mcal_image(gal_image,psf_image,reconv_psf_image,[[0,-step]]) 
+  
+  g0s = method(img0s)
+  g1p = method(img1p)
+  g1m = method(img1m)
+  g2p = method(img2p)
+  g2m = method(img2m)
+  
+  d11 = (g1p[:,0]-g1m[:,0])/(2*step)
+  d21 = (g1p[:,1]-g1m[:,1])/(2*step) 
+  d12 = (g2p[:,0]-g2m[:,0])/(2*step)
+  d22 = (g2p[:,1]-g2m[:,1])/(2*step)
+ 
+  #the matrix is correct. The transposition will swap d12 with d21 across a batch correctly.
+  R = np.array([[d11,d21],
+                [d12,d22]]).T
+  return g0s, R
+
+def get_ellipticities(img,frac=.1):
```

Comment:
> Thanks @andrevitorelli ! I made a few comments in a first pass of Review, let me know what you think.
> 
> I think that adopting this "pixels" structure will prevent us from integrating this ellipticity measurement within the rest of the autometacal framework. Do you have an example of autometacal with this ellipticity implementation?

I understand why, but the idea is to just _pixelfy_, so to speak, images when using these tf_ngmix-derived functions (ie, everything is done as usual, whenever we need gaussian moments, we quickly change the image to this and measure it). If this is properly done in tf, that shouldn't be much of an overhead, or should it?

Reply:
PS: note that this "pixel structure" is just what was previously done in `get_ellipticities`, inside the **for** loop, regarding the XX, YY, XY "meshgrids". So there will be, in a way or another, a moment when we create grids of (x,y) coords (or (u,v))...

Reply:
Yeah that's true. But I think it's just not necessary to go through this pixel representation ^^. It made sense in ngmix because they relied on numba'd for loops. But I see no reason for us to pixelfy, it just makes the code more complicated, in my opinion.

Reply:
But this can be a question of prefered coding style :-) If the pixelify approach works as efficiently and you prefer that approach, I'm ok with it.